DashAI/front/src/components/ConfigurableObject/FormInputs.jsx

```diff
@@ -0,0 +1,377 @@
+    );
+    const receivedOptions = await fetchedOptions.json();
+    setOptions(receivedOptions);
+  };
+  const getParamSchema = async () => {
+    if (selectedOption !== "") {
+      setDefaultValues({ ...defaultValues, loaded: false });
+      let fetchedParams;
+      if (selectedOption === "splits") {
+        fetchedParams = await fetch(
+          `http://localhost:8000/api/v0/select/dataloader/${selectedOption}`
+        );
+      } else {
+        fetchedParams = await fetch(
+          `${process.env.REACT_APP_SELECT_MODEL_ENDPOINT + selectedOption}`
+        );
+      }
+      const parameterSchema = await fetchedParams.json();
+      setParamSchema(parameterSchema);
+      setDefaultValues({
+        loaded: true,
+        values:
+          formDefaultValues.choice !== selectedOption
+            ? getDefaultValues(parameterSchema)
+            : formDefaultValues,
+      });
+    }
+  };
+  const handleClose = () => {
+    setOpen(false);
+  };
+  useEffect(() => {
+    getOptions(paramJsonSchema.parent);
+  }, []);
+  useEffect(() => {
+    getParamSchema();
+  }, [selectedOption]);
+  return (
+    <div key={name}>
+      <Input
+        select
+        label={name}
+        name="choice"
+        value={selectedOption}
+        onChange={(e) => setSelectedOption(e.target.value)}
+      >
+        {options.map((option) => (
+          <MenuItem key={option} value={option}>
+            {option}
+          </MenuItem>
+        ))}
+      </Input>
+      {generateTooltip(paramJsonSchema.description)}
+      <IconButton color="primary" component="label" onClick={handleButtonClick}>
+        <SettingsIcon />
+      </IconButton>
+      <Dialog open={open} onClose={handleClose}>
+        <DialogTitle>{`${selectedOption} parameters`}</DialogTitle>
+        <DialogContent key={selectedOption}>
+          {defaultValues.loaded && (
+            <Subform
+              name={name}
+              parameterSchema={paramSchema}
+              setFieldValue={setFieldValue}
+              choice={selectedOption}
+              defaultValues={defaultValues.values}
+            />
+          )}
+        </DialogContent>
+        <DialogActions>
+          <Button variant="outlined" onClick={handleClose} autoFocus>
+            Save
+          </Button>
+        </DialogActions>
+      </Dialog>
+    </div>
+  );
+}
+
+ClassInput.propTypes = {
+  name: PropTypes.string.isRequired,
+  paramJsonSchema: PropTypes.objectOf(
+    PropTypes.oneOfType([PropTypes.string, PropTypes.bool, PropTypes.object])
+  ).isRequired,
+  setFieldValue: PropTypes.func.isRequired,
+  formDefaultValues: PropTypes.objectOf(
+    PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.bool,
+      PropTypes.number,
+      PropTypes.object,
+    ])
+  ),
+};
+
+ClassInput.defaultProps = {
+  formDefaultValues: { emptyDefaultValues: true },
+};
+
+export function genInput(objName, paramJsonSchema, formik, defaultValues) {
```

Comment:
I would give this class a more descriptive name, such as `FormRenderer`, `ConfigurableObjectRenderer` or `ConfigurableObjectFormRenderer`.

```suggestion
export function FormRenderer(objName, paramJsonSchema, formik, defaultValues) {
```

If this change is made, the name of the module would also have to be changed.